### DISCUSSION TITLE
Setting cache-control to stale-while-revalidate on SSR pages

### DISCUSSION THREAD

**Discussion by joelvarty** [vercel/next.js#24221] [Help] [answered]
I'd like to be able to server-side render pages and to also set the cache-control header when hosted in Vercel.  
It looks like the cache control header is being overwritten by Vercel.  

Is there a way to control this?

Here's an example config:

```
module.exports = {
  async headers() {
    return [
      {
        source: '/post/:id*',
        headers: [
          {
            key: 'Cache-Control',
            value: 'stale-while-revalidate=600',
          },
        ],
      },
    ]
  },
}
```

**Comment by raffij** [comment]
The docs appear to say this is the behaviour https://nextjs.org/docs/api-reference/next.config.js/headers#cache-control

**Reply by joelvarty** [reply]
Agreed - I was hoping for a workaround that worked for getServerSideProps.  Lots of folks are choosing Server-Side rendering over Static rendering to avoid long build times, and also to provide better SEO.

**Comment by akshaysmurthy** [comment]
@leerob has suggested setting the cache-control header inside `getServerSideProps` - https://www.smashingmagazine.com/2021/04/incremental-static-regeneration-nextjs/#server-side-rendering

**Reply by joelvarty** [reply]
Agreed -> tested this and found it to work 100%

**Reply by akshaysmurthy** [reply]
@joelvarty Do you have this working in production? We have our app deployed on Vercel and for some strange reason, `stale-while-revalidate` does not work there. But it works well on my local machine; even with `yarn build && yarn start`

**Vercel**
<img width="399" alt="Screenshot 2021-05-02 at 10 58 11 PM" src="https://user-images.githubusercontent.com/1207578/116821948-e0839d80-ab99-11eb-9ee7-6d3fa1a83d95.png">


**Dev**
<img width="417" alt="Screenshot 2021-05-02 at 10 58 37 PM" src="https://user-images.githubusercontent.com/1207578/116821965-f002e680-ab99-11eb-9c51-b03a99c7b13f.png">

Here's my `getServerSideProps` def

```
export async function getServerSideProps({ res, params }) {
  const { regionalId } = params;
  const regional = await getRegionalInfo(regionalId);

  res.setHeader('Cache-Control', 's-maxage=300, stale-while-revalidate');

  return {
    props: {
      regional,
    },
  };
}
```

Any ideas on what might be happening here?

**Comment by joelvarty** [comment, marked as answer]
This works: 
```
export async function getServerSideProps({ params, res }: GetServerSidePropsContext<{ slug: string[] }>) {
	res.setHeader('Cache-Control', 's-maxage=300, stale-while-revalidate')


}
```

**Reply by leerob** [reply]
Amazing!

**Reply by joelvarty** [reply]
Even MORE amazing is that I've now learned about `fallback: 'blocking'` which removes the need for SSR in most cases.  The biggest question about ISR and fallback was around SEO on that first load, and blocking fixes that.  IT'S A GOOD DAY TO BE ALIVE
